hack/Makefile.debug

```diff
@@ -2,29 +2,137 @@
+DLV_CONTINUE ?= --continue
+
+build:
 	GO111MODULE=on CGO_ENABLED=0 GOFLAGS=-mod=vendor go build -o openshift-router -gcflags=all="-N -l" ./cmd/openshift-router
-	$(IMAGEBUILDER) build -t $(IMAGE):$(TAG) -f hack/Dockerfile.debug .
+	GO111MODULE=on CGO_ENABLED=0 GOFLAGS=-mod=vendor go build -o hack/remote-debug/remote-debug-helper -gcflags=all="-N -l" ./hack/remote-debug/remote-debug-helper.go
+
+build-image: build
+	$(IMAGEBUILDER) build --build-arg HAPROXY_RPMS="$(HAPROXY_RPMS)" --build-arg GITHUB_USER="$(GITHUB_USER)" -t $(IMAGE):$(TAG) -f $(DOCKERFILE_DEBUG) .
 
-push:
+push-image:
 	$(IMAGEBUILDER) tag $(IMAGE):$(TAG) $(REGISTRY)/$(IMAGE):$(TAG)
 	$(IMAGEBUILDER) push $(REGISTRY)/$(IMAGE):$(TAG)
 
 set-image:
 	oc scale --replicas 0 -n openshift-cluster-version deployments/cluster-version-operator
 	oc scale --replicas 0 -n openshift-ingress-operator deployments ingress-operator
-	oc -n openshift-ingress scale deployment --replicas=0 router-default
-	oc -n openshift-ingress patch deployment router-default -p '{"spec":{"template":{"spec":{"$$setElementOrder/containers":[{"name":"router"}],"containers":[{"imagePullPolicy":"Always","name":"router"}]}}}}'
+	oc scale --replicas 0 -n openshift-ingress deployment/router-default
+ifeq ($(DOCKERFILE_DEBUG),hack/remote-debug/Dockerfile)
+	oc apply -f hack/remote-debug/manifests
+	oc apply -f hack/remote-debug/custom-router-scc.yaml
+	@for patch in $(sort $(wildcard hack/remote-debug/patches/deployment/router-default/*.yaml)); do \
+	    echo "Applying patch $$patch"; \
+	    oc -n openshift-ingress patch deployment router-default -p "$$(cat $$patch)" --type=strategic; \
+	done
+endif
 	oc -n openshift-ingress set image deployment/router-default router=$(REGISTRY)/$(IMAGE):$(TAG)
-	oc -n openshift-ingress scale deployment --replicas=1 router-default
+	oc scale --replicas 1 -n openshift-ingress deployment/router-default
+ifeq ($(DOCKERFILE_DEBUG),hack/remote-debug/Dockerfile)
+	@echo "Waiting for the deployment/router-default to become available..."; \
+	if ! oc -n openshift-ingress rollout status deployment/router-default --timeout=180s; then \
+		echo "Deployment did not become available in time. Please check the deployment status."; \
+		false; \
+	fi
+	@max_attempts=10; \
+	attempt=1; \
+	while [ $$attempt -le $$max_attempts ]; do \
+		echo "Checking SSH access..."; \
+		if ssh -o ConnectTimeout=5 -F hack/remote-debug/ssh_config container uptime; then \
+			echo "SSH check passed."; \
+			break; \
+		else \
+			echo "SSH check failed. Restarting port-forward services and retrying..."; \
+			$(MAKE) -f hack/Makefile.debug restart-port-forward-services; \
+			attempt=$$((attempt + 1)); \
+			sleep 5; \
+		fi; \
+	done; \
+	if [ $$attempt -gt $$max_attempts ]; then \
+		echo "SSH check did not pass in time. Please check the deployment and SSH configuration."; \
+		false; \
+	fi
+endif
 
-dwim:  new-openshift-router-image push set-image
+deploy-image: build-image push-image set-image
 
 reset:
```

Comment:
If you remove the port forward service for delve, how will you debug?

Reply:
> If you remove the port forward service for delve, how will you debug?

Well, in `reset` target you remove the debug router image, you won't be able to debug anyway. Then [the preparatory steps from README](https://github.com/openshift/router/blob/ba8f66c29cc80b1cd59d93ca9f12d2d0f0b8439e/hack/remote-debug/README.md#preparatory-steps-1) will setup the port forwarding back again or [it can be set up as a prereq of `debug` target](https://github.com/openshift/router/pull/609/files#r1735867250).

Reply:
I just leave the port-forward service always running, even when not debugging. But for completeness let's clean up as best we can...

Just note that repeated invocations of reset will end up showing:

```
% make -f hack/Makefile.debug reset
for service in hack/remote-debug/oc-port-forward-delve.service hack/remote-debug/oc-port-forward-SSH.service; do \
        systemctl --user stop $(basename $service); \
        systemctl --user disable $(basename $service); \
        rm -f /home/aim/.config/systemd/user/$(basename $service); \
        systemctl --user daemon-reload; \
done
Failed to stop oc-port-forward-delve.service: Unit oc-port-forward-delve.service not loaded.
Failed to disable unit: Unit file oc-port-forward-delve.service does not exist.
Failed to stop oc-port-forward-SSH.service: Unit oc-port-forward-SSH.service not loaded.
Failed to disable unit: Unit file oc-port-forward-SSH.service does not exist.
oc scale --replicas 1 -n openshift-cluster-version deployments/cluster-version-operator
deployment.apps/cluster-version-operator scaled
oc scale --replicas 1 -n openshift-ingress-operator deployments ingress-operator
deployment.apps/ingress-operator scaled
oc delete -n openshift-ingress deployment/router-default
deployment.apps "router-default" deleted
oc delete --ignore-not-found -f hack/remote-debug/manifests
oc delete --ignore-not-found -f hack/remote-debug/custom-router-scc.yaml
rm -f -f hack/remote-debug/ssh/id_ed25519 hack/remote-debug/ssh/id_ed25519.pub
```

but that's harmless. 

Added in https://github.com/openshift/router/pull/609/commits/39e92d6ef08a0e2b69255a5fca00bbf427e93aeb.